ocaml/sdk-gen/java/main.ml

```diff
@@ -1052,6 +1061,29 @@ public class Types
   TypeSet.iter (gen_task_result_func file) !types ;
   fprintf file
     {|
+    
+    public static class BadAsyncResult extends XenAPIException
```

Comment:
If you're making other changes, I'd move this class to the beginning of the file close to BadServerResponse so we have all custom exceptions tidied up.